### DISCUSSION TITLE
Status code 500 on alerts page: all shards failed.

### DISCUSSION THREAD

**Discussion by njohnson-q** [Security-Onion-Solutions/securityonion#5780] [Unsupported Versions] [answered]
Good day,

We have an issue where we are receiving status code 500 on the alerts panel. When using developer tools I can find an error for a  url. When navigating to that url I find multiple errors. Any help is appreciated. I have posted images of the issues described below.

![image](https://user-images.githubusercontent.com/91084097/136059212-75bbc693-7f5a-4c23-998e-e1e88797e5a1.png)

![image](https://user-images.githubusercontent.com/91084097/136059822-b01d30f4-b0ae-4d08-9c86-c1b9e15a9b4f.png)

**Comment by SuiNom** [comment]
Might be worth including a little more information in this one:
1. Output of `sudo so-status`
2. Including a (sanitised) copy of the the URL you found with dev tools (might have some clues, but it all shards are failing, I'd assume anything to do with an Elasticsearch query is going to give you the same response)
3. A little background on the deployment (standalone, vs distributed, platform version, is this after an upgrade/config change, etc)

**Reply by njohnson-q** [reply]
1. Output of so-status is good on both machines
Manager:
Checking Docker status

    Docker ------------------------- [ OK ]

Checking container statuses

    so-aptcacherng ----------------- [ OK ]
    so-cortex ---------------------- [ OK ]
    so-curator --------------------- [ OK ]
    so-dockerregistry -------------- [ OK ]
    so-elastalert ------------------ [ OK ]
    so-elasticsearch --------------- [ OK ]
    so-filebeat -------------------- [ OK ]
    so-fleet ----------------------- [ OK ]
    so-grafana --------------------- [ OK ]
    so-idstools -------------------- [ OK ]
    so-influxdb -------------------- [ OK ]
    so-kibana ---------------------- [ OK ]
    so-kratos ---------------------- [ OK ]
    so-logstash -------------------- [ OK ]
    so-mysql ----------------------- [ OK ]
    so-nginx ----------------------- [ OK ]
    so-playbook -------------------- [ OK ]
    so-redis ----------------------- [ OK ]
    so-sensoroni ------------------- [ OK ]
    so-soc ------------------------- [ OK ]
    so-soctopus -------------------- [ OK ]
    so-telegraf -------------------- [ OK ]
    so-thehive --------------------- [ OK ]
    so-thehive-es ------------------ [ OK ]
    so-wazuh ----------------------- [ OK ]

Search:

Checking Docker status

    Docker ------------------ [ OK ]

Checking container statuses

    so-curator -------------- [ OK ]
    so-elasticsearch -------- [ OK ]
    so-filebeat ------------- [ OK ]
    so-logstash ------------- [ OK ]
    so-nginx ---------------- [ OK ]
    so-sensoroni ------------ [ OK ]
    so-telegraf ------------- [ OK ]
    so-wazuh ---------------- [ OK ]

2. Full url: https://SOBASEURL/api/events/?query=event.dataset:alert+AND++NOT+event.acknowledged:true+AND++NOT+event.escalated:true+AND+*+%7C+groupby+rule.name+event.module+event.severity_label&range=2021%2F10%2F06+09:22:26+AM+-+2021%2F10%2F06+09:23:26+AM&format=2006%2F01%2F02+3:04:05+PM&zone=America%2FNew_York&metricLimit=500&eventLimit=500
3. The deployment is a distributed environment, all nodes running on 2.3.70. I do not know when the exact issue came to light but I believe it was after a reboot of the manager node.

**Comment by njohnson-q** [comment, marked as answer]
Issue was resolved.

Found the following file: /opt/so/saltstack/default/salt/soc/files/soc/alerts.queries.json.

In this file added the .keyword extension to the rule.name for each query and added .keyword to event.module.

After restarting SOC the issue was resolved.
